### PR TITLE
Steven test with hash map

### PR DESCRIPTION
**Pull request by smarquez-jpg**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (Andrewmob2703-zz) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: you can probably resolve conflicts so that you get the user email and use that for merging instead of the empty string.

**Comment by Andrewmob2703-zz** [comment]
I think you can resolve conflicts so that you get the userEmail for where you currently have an empty string in the DataServlet

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: A general tip: if we need to pass a constant (String, int, double, bool, etc.) to a function argument, it's usually good to use an inlined comment like

BlogMessage message = new BlogMessage(
    messageId, tags, comment, image, sender, /*email=*/"", messageReplies, timestamp);

Then reader knows what argument is getting the "" empty string.

File: portfolio/src/test/java/com/google/sps/HashTableTest.java
Comment (peizhaoou) [inline portfolio/src/test/java/com/google/sps/HashTableTest.java]: Inlined comments applies here for all the constants and elsewhere as well.

File: portfolio/src/test/java/com/google/sps/HashTableTest.java
Comment (peizhaoou) [inline portfolio/src/test/java/com/google/sps/HashTableTest.java]: We definitely need to break the long line (elsewhere as well). The line limit is 100 characters for Java.

File: portfolio/src/test/java/com/google/sps/HashTableTest.java
Comment (peizhaoou) [inline portfolio/src/test/java/com/google/sps/HashTableTest.java]: For long constants, you don't need the (long) conversion, just append a "l" character after the number to tell the compiler to use a long integer here.

File: portfolio/src/test/java/com/google/sps/HashTableTest.java
Comment (peizhaoou) [inline portfolio/src/test/java/com/google/sps/HashTableTest.java]: "temp" in general is really not a good name for variables. Consider using "messages" or "blogMessages"

File: portfolio/src/test/java/com/google/sps/HashTableTest.java
Comment (peizhaoou) [inline portfolio/src/test/java/com/google/sps/HashTableTest.java]: Nit: Put a space between ) & {

Comment (peizhaoou) [inline portfolio/src/test/java/com/google/sps/HashTableTest.java]: Nit: This is me nitpicking and I do not have strong feeling (especially in testing code), so your call.

I see that the constant 4 is used four times. It would be good to use a local variable like:

int iterations = 4;

And then use the variable.

File: portfolio/src/test/java/com/google/sps/HashTableTest.java
Comment (peizhaoou) [inline portfolio/src/test/java/com/google/sps/HashTableTest.java:56]: About checking the equality of two objects x & y, if you define a equals() method in the object's class, you can just use Assert.assertEquals(x, y);

This would be more convenient, expressive, and concise.

**Review by peizhaoou — COMMENTED** [review]
Thanks Steven! Great to see the testing integration is working. LGTM overall, just left some comments. I think submitting after addressing them is fine with me, but I just wonder if this would need to wait for the implementation to be checked in for this to be run?

**Review by peizhaoou — APPROVED** [review]
Thanks Steven! Great to see the testing integration is working. LGTM overall, just left some comments. I think submitting after addressing them is fine with me, but I just wonder if this would need to wait for the implementation to be checked in for this to be run?

File: .gitignore
Comment (martiKallas) [inline .gitignore:1]: Before you submit this, make a single, separate change (PR) to add the .gitignore file and then merge that into this PR. That way you won't be including unneeded files.

You might be able to do that directly from the Github interface.

**Review by martiKallas — DISMISSED** [review]

**Comment by martiKallas** [comment]
Please get a review from either Andrew or Tayyaba as well!

File: portfolio/src/main/java/com/google/sps/data/BlogMessage.java
Comment (Andrewmob2703-zz) [inline portfolio/src/main/java/com/google/sps/data/BlogMessage.java]: We used 'tag' because each post can only have one tag. Currently there's no support for multiple tags per post. Please change that before merging.

**Review by Andrewmob2703-zz — APPROVED** [review]
Looks good! Just change tags to tag since we're only supporting one tag per post

**Pull request merged**
